cluster/gce/upgrade.sh

```diff
@@ -111,13 +113,57 @@ function prepare-upgrade() {
   fi
 }
 
+# Reads kube-env metadata from master and extracts value from provided key.
+#
+# Assumed vars:
+#   MASTER_NAME
+#   ZONE
+#
+# Args:
+# $1 env key to use
+function get-env-val() {
+  gcloud compute ssh --zone ${ZONE} ${MASTER_NAME} --command \
+    "curl --fail --silent -H 'Metadata-Flavor: Google' \
+      'http://metadata/computeMetadata/v1/instance/attributes/kube-env'" 2>/dev/null \
+    | grep ${1} | cut -d : -f 2 | cut -d \' -f 2
```

Comment:
It will kill the script pretty abruptly (and without any sort of error message to the user). Not too worried for the first cut, but can you add a TODO to make this more reliable?